HandsOnProject/app/src/main/java/com/example/handsonproject/MainActivity.kt

```diff
@@ -1,11 +1,160 @@
+    private var operation : Operation = Operation.MAIN_STATUS
+
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = inflate(LayoutInflater.from(this))
+        setContentView(binding.root)
+        with(binding){
+            stash.text=""
+        }
+    }
+
+
+    fun onTapB1(view: View){
+        with(binding) { input.setText(input.getText().toString() + "1") }
+    }
+    fun onTapB2(view: View){
+        with(binding) { input.setText(input.getText().toString() + "2") }
+    }
+    fun onTapB3(view: View){
+        with(binding) { input.setText(input.getText().toString() + "3") }
+    }
+    fun onTapB4(view: View){
+        with(binding) { input.setText(input.getText().toString() + "4") }
+    }
+    fun onTapB5(view: View){
+        with(binding) { input.setText(input.getText().toString() + "5") }
+    }
+    fun onTapB6(view: View){
+        with(binding) { input.setText(input.getText().toString() + "6") }
+    }
+    fun onTapB7(view: View){
+        with(binding) { input.setText(input.getText().toString() + "7") }
+    }
+    fun onTapB8(view: View){
+        with(binding) { input.setText(input.getText().toString() + "8") }
+    }
+    fun onTapB9(view: View){
+        with(binding) { input.setText(input.getText().toString() + "9") }
+    }
+    fun onTapB0(view: View){
+        with(binding) { input.setText(input.getText().toString() + "0") }
+    }
+    fun onTapBEqual(view: View){
+        with(binding){
+            if (input.text.toString() == "" && stash.text.toString() ==  "" ) return
+            when(operation){
+                Operation.PLUS -> {
+                    val res=input.text.toString().toDouble()+stash.text.toString().toDouble()
+                    input.setText(res.toString())
+                }
+                Operation.MINUS -> {
+                    val res=stash.text.toString().toDouble()-input.text.toString().toDouble()
+                    input.setText(res.toString())
+                }
+                Operation.MULTIPLY -> {
+                    val res=input.text.toString().toDouble()*stash.text.toString().toDouble()
+                    input.setText(res.toString())
+                }
+                Operation.DIVIDE -> {
+                    val divisor = input.text.toString().toDouble()
+
+                    if(divisor!=0.0) {
+                        val res=stash.text.toString().toDouble()/divisor
+                        input.setText(res.toString())
+                    }
+                    else {
+                        AlertDialog
+                            .Builder(this@MainActivity)
+                            .setTitle(getString(R.string.error_title))
+                            .setMessage(getString(R.string.error_message))
+                            .show()
+                    }
+                }
+            }
+
+            stash.setText("")
```

Comment:
@shovian As discussed in the other comment, for the stash you could use property access like this:

```
stash.text = ""
```

Reply:
it returns error

Reply:
![Uploading Screenshot from 2021-07-28 09-51-39.png…]()

Reply:
![Screenshot from 2021-07-28 09-51-39](https://user-images.githubusercontent.com/61104176/127270638-7303253b-ff16-4361-b606-045638f504a9.png)

Reply:
@shovian I am talking about the `stash`, not the `input`. See: https://github.com/shovian/HandsOnProject/pull/5#discussion_r677939662

Reply:
it works!

Reply:
great